includes/functions.inc.php

```diff
@@ -404,9 +404,14 @@ function pagination($page_count,$page,$browse_range=3,$show_last=1)
  */
 function make_link($string)
```

Comment:
I don't like this solution. I would prefer to define the target-value in `$settings['link_open_target']`, i.e. `_blank` instead of a (pseudo) boolean value. In this case, one can specify every possible frame. Furthermore, internal should handled differently the external ones but, currently, these links will also open in a new window. Please note, the `[msg]` is not used anymore.

Reply:
> I would prefer to define the target-value in $settings['link_open_target'], i.e. _blank instead of a (pseudo) boolean value. In this case, one can specify every possible frame.

I had the same idea but I can not find a source about what happens when one specified a unknown value for target. Will the attribute get ignored (would be the easiest solution)?

Side note: In the past I was able to search the old SelfHTML documentation where questions like this was answered. Nowadays the documentation focuses more on tasks for everyday work but not about such special questions. I think I have the old documentation somewhere in a backup (Matthias Apsel (one of the club members) gave me a copy a long time ago but the copy vanished later).

Reply:
> Furthermore, internal links should handled differently then external ones but, currently, these links will also open in a new window.

Ah, I see. Thank you for the hint. I'll add a check for non internal URLs.

> Please note, the `[msg]` is not used anymore.

Err, … no, not complete. Yes, with a more or less up-to-date version you get no new occurences of `msg`. No, in a forum that was initially installed before we removed this bb-code you will find `[msg]` with a great chance.

Reply:
> Err, … no, not complete. Yes, with a more or less up-to-date version you get no new occurences of `msg`. No, in a forum that was initially installed before we removed this bb-code you will find `[msg]` with a great chance.

The `msg`-Tag is already valid and can be used to link to a posting but, if one use the bb-code butten, the `url` tag is used. Beside `url` the forum also supports `link`. For that reason, `msg` can still occur...

Reply:
> I had the same idea but I can not find a source about what happens when one specified a unknown value for target. Will the attribute get ignored (would be the easiest solution)?

No, afaik, a new window is opened, cf. [eigener Fenstername für vorhandenes oder **neues Fenster**](https://wiki.selfhtml.org/wiki/Referenz:HTML/Attribute/target).

Reply:
I found the former SELFHTML documentation. Is states: "Anstelle von target="_blank" können Sie auch target="Halligalli" oder sonst irgendeinen Unsinn notieren. Wenn es kein Fenster mit diesem Namen gibt, öffnen die Web-Browser das Verweisziel in einem neuen Browser-Fenster.", cf. [Zielfenster für Verweise bestimmen](https://www2.informatik.hu-berlin.de/Themen/www/selfhtml/html/verweise/definieren.htm#zielfenster)

Reply:
> I found the former SELFHTML documentation. Is states …

Thank you again for your hint.

Reply:
> Side note: In the past I was able to search the old SelfHTML documentation where questions like this was answered. Nowadays the documentation focuses more on tasks for everyday work but not about such special questions. I think I have the old documentation somewhere in a backup (Matthias Apsel (one of the club members) gave me a copy a long time ago but the copy vanished later).

Are you interested in the 8.43MB selfhtml-8.1.2.chm of May 2007? Let me know.

Reply:
> Are you interested in the 8.43MB selfhtml-8.1.2.chm of May 2007?

No, I found my own copy of SelfHTML 8.1.2 in HTML format (just now). But thank you for your offer.